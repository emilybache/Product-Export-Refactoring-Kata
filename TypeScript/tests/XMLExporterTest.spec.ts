
describe("XMLExporter", () => {

    it("Something", () => {

        // TODO: implement this test

        // expect(xml).toMatchSnapshot(); // Use snapshot testing
    });

});
