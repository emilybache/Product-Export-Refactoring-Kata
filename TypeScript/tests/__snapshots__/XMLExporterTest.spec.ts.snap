// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XMLExporter exportFull 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<orders>
<order id='1234'>
<product id='EVENT02'
 stylist='Celeste Pulchritudo'
>
<price currency='USD'>149.99</price>
Makeover
</product>
</order>
<order id='1235'>
<product id='LIPSTICK01'
 weight='30'
>
<price currency='USD'>14.99</price>
Cherry Bloom
</product>
</order>
</orders>"
`;

exports[`XMLExporter exportHistory 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<orderHistory createdAt='2018-09-20T00:00Z'>
<order date='2018-08-31T12:00Z' totalDollars='149.99'>
<product id='EVENT02'>Makeover</product>
</order>
<order date='2017-08-31T12:00Z' totalDollars='14.99'>
<product id='LIPSTICK01'>Cherry Bloom</product>
</order>
</orderHistory>"
`;

exports[`XMLExporter exportStore 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<store name='Nordstan'>
<product id='LIPSTICK01'
 weight='30'
>
<price currency='USD'>14.99</price>
Cherry Bloom
</product>
<product id='EVENT01'
 location='Nordstan'
>
<price currency='USD'>119.99</price>
Eyeshadow Masterclass
</product>
<product id='EVENT02'
 location='Nordstan'
>
<price currency='USD'>149.99</price>
Makeover
</product>
</store>"
`;

exports[`XMLExporter exportTaxDetails 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<orderTax>
<order date='2018-08-31T12:00Z'>
<product id='EVENT02'>Makeover</product>
<orderTax currency='USD'>57.50</orderTax>
</order>
<order date='2017-08-31T12:00Z'>
<product id='LIPSTICK01'>Cherry Bloom</product>
<orderTax currency='USD'>12.62</orderTax>
</order>
70.12
</orderTax>"
`;
